packages/contracts/contracts/MarketRegistry.sol

```diff
@@ -656,6 +670,17 @@ contract MarketRegistry is
         }
     }
 
+    function setMarketPaymentCycleType(
+        uint256 _marketId,
+        PaymentCycleType _paymentCycleType
+    ) public ownsMarket(_marketId) {
+        if (_paymentCycleType != paymentCycleType[_marketId]) {
+            paymentCycleType[_marketId] = _paymentCycleType;
+
+            emit SetMarketPaymentCycleType(_marketId, _paymentCycleType);
```

Comment:
lets merge the functions `setPaymentCycleDuration` and `setMarketPaymentCycleType` into a single function `setPaymentCycle` so we can run some additional checks on both values at the same time

i.e.
```solidity
    function setPaymentCycle(uint256 _marketId, uint32 _type, uint32 _duration)
        public
        ownsMarket(_marketId)
    {
        require(
            (_type == PaymentCycleType.Seconds) || 
            (_type == PaymentCycleType.Monthly && _duration == 0),
            "monthly payment cycle value must be 0"
        );
        Marketplace storage market = markets[_marketId];
				uint32 duration = _type == PaymentCycleType.Seconds ? _duration : 30 days;
        if (_type != market.paymentCycleType || duration != market.paymentCycleDuration) {
            markets[_marketId].paymentCycleType = _type;
            markets[_marketId].paymentCycleDuration = duration;

            emit SetPaymentCycle(_marketId, _type, duration);
        }
```